src/components/layout/Stack/Stack.tsx

```diff
@@ -28,89 +23,64 @@ export interface StackProps {
    */
   align?: Property.JustifyContent;
   /**
-   * Index of element after which the Stack is splitted. Leave 'undefined' for no splitting.
+   * Index of element after which the Stack is splitted. Leave 'undefined' for no splitting
    */
   splitAt?: number;
   /**
    * Should apply gap recursively (on nested levels)
    */
   isRecursive?: boolean;
-  className?: string;
   children: ReactNode;
-  as?: ElementType;
 }
-type StackRootProps = {
-  $gap: StackProps['gap'];
-  $justify: StackProps['justify'];
-  $align: StackProps['align'];
-  $splitAt: StackProps['splitAt'];
-  $isRecursive: StackProps['isRecursive'];
-};
 
-const StackRoot = styled.div<StackRootProps>`
+const Stack = styled.div
+  .withConfig<StackProps>({
+    shouldForwardProp: (propName) =>
+      !['gap', 'justify', 'align', 'splitAt', 'isRecursive', 'theme'].includes(
+        propName,
+      ),
+  })
```

Comment:
@silvioprog It's important to filter our any property used for styling (used inside styled-component definition) to prevent leaking those properties into DOM. The `theme` property is used internally and should be always filtered out.